cns/grpc/server.go

```diff
@@ -54,13 +60,22 @@ func (s *Server) Start() error {
 
 	grpcServer := grpc.NewServer()
 	pb.RegisterCNSServer(grpcServer, s.CnsService)
+	healthgrpc.RegisterHealthServer(grpcServer, s.HealthServer)
 
 	// Register reflection service on gRPC server.
 	reflection.Register(grpcServer)
 
+	// Set initial health status
+	s.HealthServer.SetServingStatus("", healthgrpc.HealthCheckResponse_SERVING)
```

Comment:
looks like you're setting the health server to "serving" before you have actually started the primary server, which would make that false?